taroscript/send_test.go

```diff
@@ -301,201 +312,100 @@ func createGenesisProof(t *testing.T, state *spendData) {
 	state.asset2GenesisProof = asset2GenesisProof
 }
 
-func createSpend(t *testing.T, state *spendData, spend taroscript.SpendDelta,
-	full bool) (psbt.Packet, [32]byte, taroscript.SpendDelta,
-	taroscript.SpendCommitments) {
-
-	var (
-		spendAddress     address.Taro
-		receiverStateKey [32]byte
-	)
+func createPacket(addr address.Taro, prevInput asset.PrevID,
+	state spendData, inputSet commitment.InputSet) *taropsbt.VPacket {
 
-	if full {
-		spendAddress = state.address2
-		receiverStateKey = state.address2StateKey
-	} else {
-		spendAddress = state.address1
-		receiverStateKey = state.address1StateKey
+	inputAsset := inputSet[prevInput]
+	input := &taropsbt.VInput{
+		PrevID: prevInput,
 	}
+	outputs := []*taropsbt.VOutput{{
+		Amount: inputAsset.Amount - addr.Amount,
+		ScriptKey: asset.NewScriptKey(
+			&state.spenderScriptKey,
+		),
+		AnchorOutputIndex:       0,
+		IsChange:                true,
+		AnchorOutputInternalKey: &state.spenderPubKey,
+	}, {
+		Amount:                  addr.Amount,
+		ScriptKey:               asset.NewScriptKey(&addr.ScriptKey),
+		AnchorOutputIndex:       receiverExternalIdx,
+		AnchorOutputInternalKey: &addr.InternalKey,
+	}}
+
+	vPacket := &taropsbt.VPacket{
+		Input:       input,
+		Outputs:     outputs,
+		ChainParams: &address.RegressionNetTaro,
```

Comment:
The test addresses are all for mainnet, so we may want to add a network check to `taropsbt.FromAddress()` and elsewhere.

Reply:
Looks like these checks are partially added in the follow-up PR, but they wouldn't apply in the tests since we create `VPackets` directly.

I think this could be replaced with `FromAddress` for the positive test cases.

Reply:
Ah good point about the mismatch. I fixed it in the tests. Though I don't know what value to check against in `FromAddress()`? Also, `FromAddress()` does slightly less than this does, so it would be another partial refactor which I'd like to avoid at this point.

Reply:
These changes work, I meant that both `FromAddress()` and the test logic to make packets should copy the chainparams from the address.